src/mapgen/mapgen_carpathian.cpp

```diff
@@ -80,6 +81,8 @@ MapgenCarpathian::MapgenCarpathian(
 	noise_hills         = new Noise(&params->np_hills,         seed, csize.X, csize.Z);
 	noise_ridge_mnt     = new Noise(&params->np_ridge_mnt,     seed, csize.X, csize.Z);
 	noise_step_mnt      = new Noise(&params->np_step_mnt,      seed, csize.X, csize.Z);
+	if (spflags & MGCARPATHIAN_RIVERS)
+		noise_rivers    = new Noise(&params->np_rivers,        seed, csize.X, csize.Z);
```

Comment:
`noise_rivers` stays uninitialized in the else-case. Please adjust that here on in the header file to assign it with `nullptr`.

Reply:
Hmm, the other mapgens do the same:
https://github.com/minetest/minetest/blob/5b8363af0084786a7e949ae8fa16bde046f8ffda/src/mapgen/mapgen_v7.cpp#L91

Reply:
I see. My suggestion was to increase the safety in case the uninitialized noise pointer is accessed under certain conditions in the code. Even when it's done the same way somewhere else it does not mean that it's a good idea.

Reply:
Ok will do.

Reply:
Done.